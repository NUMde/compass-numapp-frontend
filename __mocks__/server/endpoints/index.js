import login from './login';
import getUser from './getUser';
import getLanguages from './getLanguages';
import updateLanguage from './updateLanguage';
import updateToken from './updateToken';
import getQuestionnaire from './getQuestionnaire';
import sendQuestionnaire from './sendQuestionnaire';
import report from './report';

export default [
  login,
  getUser,
  getLanguages,
  updateLanguage,
  updateToken,
  getQuestionnaire,
  sendQuestionnaire,
  report,
];
