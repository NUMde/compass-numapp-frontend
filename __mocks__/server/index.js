export { default as server } from './server';
