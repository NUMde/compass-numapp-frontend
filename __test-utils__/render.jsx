/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';

import UserReducer from '../src/store/user.slice';
import GlobalsReducer from '../src/store/globals.slice';
import QuestionnaireReducer from '../src/store/questionnaire.slice';

const testReducer = combineReducers({
  User: UserReducer,
  Globals: GlobalsReducer,
  Questionnaire: QuestionnaireReducer,
});

function renderWithRedux(
  ui,
  {
    initialState = {
      User: UserReducer.initialState,
      Globals: GlobalsReducer.initialState,
      Questionnaire: QuestionnaireReducer.initialState,
    },
    store = createStore(testReducer, initialState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react-native';

export { renderWithRedux };
