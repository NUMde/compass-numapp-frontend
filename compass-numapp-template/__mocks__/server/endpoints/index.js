import login from './login';
import getLanguages from './getLanguages';
import updateLanguage from './updateLanguage';
import updateToken from './updateToken';
import getQuestionnaire from './getQuestionnaire';
import sendQuestionnaire from './sendQuestionnaire';

export default [
  login,
  getLanguages,
  updateLanguage,
  updateToken,
  getQuestionnaire,
  sendQuestionnaire,
];
