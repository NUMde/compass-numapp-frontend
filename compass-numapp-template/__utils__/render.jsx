import React from 'react';
import { Provider } from 'react-redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react-native';
import { buildStore } from './mockStore';

let store = null;

export function renderWithRedux(subComponent, state = {}) {
  if (!store) {
    store = buildStore(state);
  }

  const queries = render(<Provider store={store}>{subComponent}</Provider>);

  return {
    ...queries,
    store,
  };
}

export function initRenderModule(initialState = {}) {
  store = buildStore(initialState);

  return store;
}
