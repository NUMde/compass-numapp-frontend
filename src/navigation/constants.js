/**
 * enum for all route names
 */
const Routes = {
  LANDING: 'Landing',
  LOGIN: 'Login',
  CHECK_IN: 'CheckIn',
  SURVEY: 'Survey',
  ABOUT: 'About',
  WEBVIEW: 'Webview',
  LEGAL_INFORMATION: 'LegalInformation',
};

/**
 * enum for all stacks
 */
const Stacks = {
  SIGNED_IN: 'SignedIn',
  SIGNED_OUT: 'SignedOut',
};

export { Routes, Stacks };
