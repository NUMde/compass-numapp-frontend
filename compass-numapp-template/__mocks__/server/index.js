import { setupServer } from 'msw/node';
import endpoints from './endpoints';

export default setupServer(...endpoints);
