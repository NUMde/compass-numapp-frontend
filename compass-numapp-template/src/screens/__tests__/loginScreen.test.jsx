import React from 'react';

import { renderWithRedux, waitFor } from '__test-utils__/render';
import server from '__mocks__/server';

import { appConfig } from '~config';

import { Stacks, Routes } from '~navigation/constants';
import LoginScreen from '../loginScreen';

describe('LoginScreen', () => {
  beforeAll(() => {
    server.listen();
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    server.close();
  });

  it('should render the login screen', () => {
    const navigate = jest.fn();
    const goBack = jest.fn();
    appConfig.automateQrLogin = false;
    const { toJSON } = renderWithRedux(
      <LoginScreen navigation={{ navigate, goBack }} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should automatically log in when enabled in config', async () => {
    const navigate = jest.fn();
    const goBack = jest.fn();
    appConfig.automateQrLogin = true;
    renderWithRedux(<LoginScreen navigation={{ navigate, goBack }} />);
    await waitFor(() =>
      expect(navigate).toHaveBeenCalledWith(Stacks.SIGNED_IN, {
        screen: Routes.CHECK_IN,
      }),
    );
  });
});
