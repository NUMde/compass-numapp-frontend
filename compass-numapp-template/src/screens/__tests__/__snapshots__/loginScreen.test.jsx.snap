// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render the login screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F9FA",
      "height": "100%",
    }
  }
  testID="LoginScreen"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "flexDirection": "column",
        "height": 354.06976744186045,
        "paddingTop": 30,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 354.06976744186045,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 20,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityHint="Jumps to previous page."
          accessibilityLabel="back"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="banner_back_btn"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#EB576A",
                    "fontSize": 48.83720930232558,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              Û∞Åç
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          accessibilityRole="header"
          numberOfLines={1}
          style={
            Object {
              "color": "#1F2532",
              "fontFamily": "IBMPlexSans-SemiBold",
              "fontSize": 33.75,
              "lineHeight": 43.125,
            }
          }
        >
          Login
        </Text>
      </View>
      <View
        style={
          Object {
            "height": 26,
            "width": 26,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "bottom": 13.081395348837209,
            "maxHeight": 308.781773931855,
            "maxWidth": 650,
            "position": "relative",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flexGrow": 1,
          }
        }
        enableOnAndroid={true}
        onScroll={[Function]}
        scrollEventThrottle={16}
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "width": "100%",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "backgroundColor": "#F8F9FA",
                "height": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F8F9FA",
                  "flex": 1,
                  "height": "100%",
                }
              }
            >
              <View
                testID="scannerWrapper"
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                      Object {
                        "height": 750,
                        "marginTop": 0,
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flex": 2,
                          "justifyContent": "center",
                          "width": 750,
                        },
                        undefined,
                      ]
                    }
                  />
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "fontSize": 16,
                            "textAlign": "center",
                          }
                        }
                      >
                        ...
                      </Text>
                    </View>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flex": 2,
                          "justifyContent": "center",
                          "width": 750,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#1F2532",
                    "fontFamily": "IBMPlexSans",
                    "fontSize": 30,
                    "lineHeight": 39.375,
                    "marginBottom": 0,
                    "marginLeft": 122.09302325581395,
                    "marginRight": 122.09302325581395,
                    "marginTop": 52.325581395348834,
                    "textAlign": "center",
                  }
                }
              >
                Please point the camera onto the qr-code.
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
